x509util/certificate_request.go

```diff
@@ -25,6 +52,7 @@ type CertificateRequest struct {
 	SANs               []SubjectAlternativeName `json:"sans"`
 	Extensions         []Extension              `json:"extensions"`
 	SignatureAlgorithm SignatureAlgorithm       `json:"signatureAlgorithm"`
+	ChallengePassword  string                   `json:"-"`
```

Comment:
Instead of supporting this specific property like this, it might be an option to support it as a slice of `Attributes`? Then the `ChallengePassword` could be one specific `Attribute`. Processing it would be similar to `[]Extension`, but with some custom logic for the additional (supported) attribute OIDs and values.

Probably good for a future iteration, though.

Reply:
A nice to have: support the challenge password in `NewCertificateRequestFromX509` too?

Reply:
I was focusing on adding minimal support to create CSRs with it. I'm not focusing on parsing them, this can go in a future PR. Parsing requires a little bit more effort, but it's not hard.